R/eval-walk.R

```diff
@@ -113,6 +121,12 @@ ensure_named <- function(pos,
   pos
 }
 
+check_empty <- function(x, allow_empty = TRUE, call = caller_env()) {
+  if (!allow_empty && length(x) == 0) {
+    abort("Must select at least one item", call = call)
```

Comment:
```suggestion
    abort("Must select at least one item.", call = call)
```

Adding the full stop

An alternative error that is similar to the `allow_rename` error is something like:

```
Can't generate an empty selection in this context.
```

Reply:
The seems a bit vague to me — for the tidyr cases, I'd prefer (e.g.) `` `values_from` must select at least one item. ``

Reply:
That seems fine.

So maybe an error arg of some kind would be useful. I think it would probably be useful in a few places though, not just with `allow_empty`, so maybe it is worth doing that in a separate PR